Spigot-API-Patches/0193-Add-PlayerAttackEntityCooldownResetEvent.patch

```diff
@@ -0,0 +1,87 @@
+From 8569d485d64336a0b3357f905a5ba95940d8f19b Mon Sep 17 00:00:00 2001
+From: nossr50 <nossr50@gmail.com>
+Date: Thu, 26 Mar 2020 19:30:58 -0700
+Subject: [PATCH] Add PlayerAttackEntityCooldownResetEvent
+
+
+diff --git a/src/main/java/com/destroystokyo/paper/event/player/PlayerAttackEntityCooldownResetEvent.java b/src/main/java/com/destroystokyo/paper/event/player/PlayerAttackEntityCooldownResetEvent.java
+new file mode 100644
+index 00000000..05a0f2f5
+--- /dev/null
++++ b/src/main/java/com/destroystokyo/paper/event/player/PlayerAttackEntityCooldownResetEvent.java
+@@ -0,0 +1,72 @@
++package com.destroystokyo.paper.event.player;
++
++import org.bukkit.entity.Entity;
++import org.bukkit.entity.Player;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.player.PlayerEvent;
++import org.jetbrains.annotations.NotNull;
++
++/**
++ * Called when processing a player's attack on an entity when the player's attack strength cooldown is reset
++ */
++public class PlayerAttackEntityCooldownResetEvent extends PlayerEvent implements Cancellable {
++
++    private final float cooledAttackStrength;
++    private boolean cancel = false;
++    private static final HandlerList handlers = new HandlerList();
++    @NotNull private final Entity attackedEntity;
++
++    public PlayerAttackEntityCooldownResetEvent(@NotNull Player who, @NotNull Entity attackedEntity, float cooledAttackStrength) {
++        super(who);
++        this.attackedEntity = attackedEntity;
++        this.cooledAttackStrength = cooledAttackStrength;
++    }
++
```

Comment:
you need
```
    @NotNull
    public static HandlerList getHandlerList() {
        return handlers;
    }
```
as well somewhere in here

Reply:
That's strange, I'm pretty sure I had that in the project. I'll look over it later.

Reply:
@Spottedleaf getHandlers() is present, I just checked.

Reply:
You need both, this is how it should look. https://imgur.com/a/GNDYdzy. You're missing the latter.

Reply:
Ah thanks, I misread then. I'll fix it.

Reply:
Added